flow360/component/flow360_params/time_stepping.py

```diff
@@ -105,6 +120,17 @@ class UnsteadyTimeStepping(BaseTimeStepping):
     model_type: Literal["Unsteady"] = pd.Field("Unsteady", alias="modelType", const=True)
     physical_steps: Optional[PositiveInt] = pd.Field(alias="physicalSteps")
     time_step_size: Optional[TimeType.Positive] = pd.Field(alias="timeStepSize")
+    CFL: Optional[Union[RampCFL, AdaptiveCFL]] = pd.Field(
+        displayed="CFL", options=["Ramp CFL", "Adaptive CFL"]
+    )
+
+    # pylint: disable=C0103
+    def __init__(self, **data):
+        super().__init__(**data)
+        if self.CFL is None or (isinstance(self.CFL, AdaptiveCFL) and self.CFL.asked_for_default()):
+            self.CFL = AdaptiveCFL.default_unsteady()
+        elif isinstance(self.CFL, RampCFL) and self.CFL.asked_for_default():
+            self.CFL = RampCFL.default_unsteady()
```

Comment:
a more pydantic way would be to use validators instead of constructor, for examples:
```python
    @pd.root_validator(pre=True)
    def set_default_cfl(cls, values):
        print(values)
        if "CFL" in values:
            cfl = values['CFL']
            if isinstance(cfl, RampCFL):
                cfl_as_dict = cfl.dict(exclude_unset=True)
                print(cfl_as_dict)
        return values

```

here you can access what user provided, for example, user might specify only ramp->final but the rest will be default. `cfl_as_dict` is exactly what user provided. You want to set  values['CFL'] to default and overwrite it with `cfl_as_dict`. Note that type(values['CFL']) should be `RampCFL` when you return as no more validation will happen on this object.

For adaptive default, I would put them directly in CFL property definition:
```python
CFL: Optional[Union[RampCFL, AdaptiveCFL]] = pd.Field(AdaptiveCFL(min=..., max=..., ), displayed="CFL", options=["Ramp CFL", "Adaptive CFL"])
```

Reply:
Sure I changed with root_validator way. But we still go through pydantic validation since we have pre=True right?